backend/canisters/event_relay/impl/src/lib.rs

```diff
@@ -46,20 +55,44 @@ impl RuntimeState {
 #[derive(Serialize, Deserialize)]
 struct Data {
     pub push_events_whitelist: HashSet<Principal>,
+    pub events_sink_client: event_sink_client::Client<CdkRuntime>,
+    pub event_deduper: EventDeduper,
     pub cycles_dispenser_canister_id: CanisterId,
+    pub salt: [u8; 32],
     pub rng_seed: [u8; 32],
     pub test_mode: bool,
 }
 
 impl Data {
-    pub fn new(push_events_whitelist: HashSet<Principal>, cycles_dispenser_canister_id: CanisterId, test_mode: bool) -> Data {
+    pub fn new(
+        push_events_whitelist: HashSet<Principal>,
+        events_sink_canister_id: CanisterId,
+        cycles_dispenser_canister_id: CanisterId,
+        test_mode: bool,
+    ) -> Data {
         Data {
             push_events_whitelist,
+            events_sink_client: event_sink_client::ClientBuilder::new(events_sink_canister_id, CdkRuntime::default()).build(),
+            event_deduper: EventDeduper::default(),
             cycles_dispenser_canister_id,
+            salt: [0; 32],
             rng_seed: [0; 32],
             test_mode,
         }
     }
+
+    pub fn obfuscate_user(&self, user: String) -> String {
+        // We only want to obfuscate userId principals, so if the string is not a principal we return it as is
+        if Principal::from_text(&user).is_err() {
```

Comment:
Possibly an unnecessary restriction

Reply:
Yeah I was just thinking if we ever say user = ProposalsBot or something like that

Reply:
I'll remove the restriction for now since we may never need it